plugin/consul/plugin.go

```diff
@@ -42,109 +43,153 @@
 		Version: "0.0.1",
-		Features: PluginFeatures{
+		Features: plugin.PluginFeatures{
 			Target: "yes",
 			Store:  "no",
 		},
 		Example: `
 {
-  "host"     : "consul-endpoint",      # Optional hostname, with or without the
-                                       # http:// or https:// URL scheme.
+  "consul"      : "/var/vcap/packages/consul/bin/consul"     # Path to the consul binary
+  "address"     : "consul.service.consul:8200",              # Optional hostname, with or without the http:// or https:// URL scheme.
 
-  "skip_ssl_validation" : false        # Skip certificate verification (not recommended)
+  "ca-path"     : "/var/vcap/jobs/consul/consul/ca.cert"     # Path to the CA certificate to use for TLS when communicating with Consul via https.
+  "client-cert" : "/var/vcap/jobs/consul/consul/consul.cert" # Path to a client cert file to use for TLS when 'verify_incoming' is enabled.
+  "client-key"  : "/var/vcap/jobs/consul/consul/consul.key"  # Path to a client key file to use for TLS when 'verify_incoming' is enabled.
 
-  "username" : "basic-auth-username",  # Optional username and password
-  "password" : "basic-auth-password"   # for HTTP Basic Authentication.
 }
 `,
 		Defaults: `
 {
-  "host" : "http://127.0.0.1:8500"
+  "consul"  : "/var/vcap/packages/consul/bin/consul"
+  "address" : "http://127.0.0.1:8500"
 }
 `,
 	}
 
-	Run(p)
+	plugin.Run(p)
 }
 
-type ConsulPlugin PluginInfo
+type ConsulPlugin plugin.PluginInfo
 
-type ConsulConnectionInfo struct {
-	Host     string
-	Port     string
-	User     string
-	Password string
-	Bin      string
-	Database string
+type ConsulConfig struct {
+	Consul     string
+	Address    string
+	CaPath     string
+	ClientCert string
+	ClientKey  string
 }
 
-func (p ConsulPlugin) Meta() PluginInfo {
-	return PluginInfo(p)
+func (p ConsulPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
 }
 
-func (p ConsulPlugin) Validate(endpoint ShieldEndpoint) error {
+func getConsulConfig(endpoint plugin.ShieldEndpoint) (*ConsulConfig, error) {
+	consul, err := endpoint.StringValueDefault("consul", DefaultConsul)
+	if err != nil {
+		return nil, err
+	}
+
+	address, err := endpoint.StringValueDefault("address", DefaultAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	ca_path, err := endpoint.StringValueDefault("ca-path", "")
+	if err != nil {
+		return nil, err
+	}
+
+	client_cert, err := endpoint.StringValueDefault("client-cert", "")
+	if err != nil {
+		return nil, err
+	}
+
+	client_key, err := endpoint.StringValueDefault("client-key", "")
+	if err != nil {
+		return nil, err
+	}
+
+	return &ConsulConfig{
+		Consul:     consul,
+		Address:    address,
+		CaPath:     ca_path,
+		ClientCert: client_cert,
+		ClientKey:  client_key,
+	}, nil
+}
+
+func (p ConsulPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
 	var (
 		s    string
-		b    bool
 		err  error
 		fail bool
 	)
 
-	s, err = endpoint.StringValueDefault("host", "")
+	s, err = endpoint.StringValueDefault("consul", "")
```

Comment:
Is this a backwards-breaking change?  Will existing configurations with a `host` key continue to work once this code is merged in?